config/endpoint/endpoint.go

```diff
@@ -363,6 +363,17 @@ func (e *Endpoint) call(result *Result) {
 		}
 		result.Duration = time.Since(startTime)
 	} else if endpointType == TypeSSH {
+		if len(e.SSHConfig.Username) == 0 && len(e.SSHConfig.Password) == 0 {
```

Comment:
Can you add a comment below this line? 
Something like `// If there's no username/password specified, attempt to validate just the SSH banner`

(I'm on mobile, otherwise I'd have made the change myself)